### PR TITLE
essay: ZKP Computation Verification.

### PR DESCRIPTION
**Pull request by vivi365**
# Assignment Proposal

## Title

_ZKP Computation Verification: Scaling Ethereum using Zero-Knowledge Proofs_

## Names and KTH ID

- Vivi Andersson (vivia@kth.se)

## Deadline

- Task 1

## Category

- Essay

## Description

As the Ethereum network has evolved, scalability issues have become eminent. One consequence of increased traffic on the Ethereum blockchain is higher gas prices and transaction congestion. Ethereum has adopted various approaches to alleviate these issues, including on-chain and off-chain approaches.

One notable off-chain scaling approach is layer-2 scaling, which involves outsourcing computation. Recently, Zero-Knowledge proofs have played an increasing role in Layer-2 scaling solutions by providing a method for verifying computations made off-chain. These proofs play a key role in verifying computations conducted off-chain, contributing significantly to the effectiveness of Layer-2 scaling solutions.

In this essay, I aim to explore zero-knowledge proofs for computation verification in the context of Ethereum scaling solutions. Different ZKP approaches, their implementations and a discussion of challenges and possibilities will be included.

**Relevance**

Scaling solutions are essential for a wider adoption of blockchain technology. By understanding blockchain scaling challenges and the mechanisms behind state-of-art solutions we can better understand the blockchain landscape, and possibly, in the future contribute to the development of this field.

### CODE REVIEW

**Comment by javierron** [comment]
Hello @anderssonvivi. Thanks for the submission

**Comment by ChristoferVikstroem** [comment]
I received Vivi's essay yesterday (12th of December) and here is my feedback on: ZKP Computation Verification: Scaling Ethereum using Zero-Knowledge Rollups. 

“I certify that generative AI, incl. ChatGPT, has not been used to write this feedback. Using generative AI without permission is considered academic misconduct.”

**High-level strengths:** The overall essay is well-structured, has good coverage and motivation of the subject matter, in particular the layer 2 solution for the scaling issue on- and off-blockchain, and has technical depth. The essay technically explains zero-knowledge implementations for scaling Ethereum, including the state of the art, which is fundamental to the essay topic. 
The use of figures is clear and purposeful, giving the reader a visual aid to conceptually understand the relationship between Layer 1 and Layer 2. The inclusion of the appendix with an example implementation is a great way to engage the reader, allowing them to either interact with the referenced repository and code or just observe and analyze the code to get a deeper understanding of how the process of verification takes place. 
**High-level weaknesses:** Besides some detail issues, for instance, the wrong course code and duplicate discussion sections (one is currently empty), some inclusions could help make the essay stronger. Ways to strengthen the essay would be for the discussion to include alternative scaling solutions or contrast the findings with papers that tackled similar problems. The paper Scaling Ethereum 2.0s Cross-Shard Transactions with Refined Data Structures discusses a cross-shard communication method that presents results of doubled transactions per block by compression. https://doi.org/10.3390/cryptography6040057
Although the essay has a brief section on the state of the art and introduces zk-STARKs and the zk-SNARK protocol, the essay would benefit by mentioning sharding designs such as danksharding (https://ethereum.org/en/roadmap/danksharding/). This would expand the section and provide more meaningful context to the issue at hand, which is scaling Ethereum by making transactions on Layer 2 as cheap as possible. 

The general feedback points mostly concern the discussion, conclusion, and an expansion of the state of the art. 
-	Mentioning or introducing relevant Ethereum designs, like danksharding, would give further context to the state of the art in scalability. 
-	Including other scaling alternatives in the discussion would make the discussion section more interesting as it would allow more contrast and discussion from the author. 
-	Expanding the conclusion by reiterating on the specific topic and areas discussed in the essay would make the “take-home”-message louder and clearer for the reader. It would also be interesting to hear suggestions for future potential research or development. This would further help justify the essay and hopefully motivate readers to continue the work that was presented. 
-	The mentioned tradeoffs are clearly stated in the essay and specific examples are presented. Although the comparison between optimistic rollups with zk rollups is made, I believe that either a deeper or broader explanation could have been made. To make it deeper, the essay could have made more explicit comparisons with alternative scaling solutions/options, which could provide insight into the decision-making process. To make it broader, the essay could have discussed how the presented tradeoffs fit into the ecosystem of blockchain, which would help to contextualize it into a “real-life”-context. 

To summarize, I think that the essay has a good structure, is easy to follow and understand, well well-referenced, and overall gives a good introduction to ZKP computation verification. The weaknesses are not severe and would only enhance the essay by giving it more body to work with, like depth in the discussion or suggestions on where to go from here. The structure issues, e.g. course code and duplicate discussion section header, and very minor and easy to fix. 

Good job!

**Comment by vivi365** [comment]
Thank you for the feedback @ChristoferVikstroem!
